npm/vite-dev-server/src/resolveConfig.ts

```diff
@@ -31,10 +32,48 @@ export const createConfig = async ({ options, viteConfig: viteOverrides = {} }:
     Falling back to Vite\'s defaults.`)
   }
 
+  debug(`Resolving to Vue, React, and Mount Utils packages here:`,
+    join(binaryRoot, 'npm/vue'),
+    join(binaryRoot, 'npm/react'),
+    join(binaryRoot, 'npm/mount-utils'))
+
   const config = {
     root,
     base: `/${options.config.namespace}/src/`,
     configFile,
+    server: {
+      fs: {
+        // https://vitejs.dev/config/#server-fs-allow
+        //
+        // By default, Vite only allows dependency scanning within the user's own project directories.
+        // Cypress will load npm/* projects as external modules from the binary's location.
+        // Because the Cypress binary is not located within the user's node_modules,
+        // Vite needs to be explicitly allowed to traverse the cypressBinaryRoot directory.
+        //
+        // If `resolve.alias` did not map to a cypressBinaryRoot outside of the user's own project,
+        // then this line would not be necessary.
+        allow: [
+          searchForWorkspaceRoot(process.cwd()),
+          binaryRoot,
+        ],
```

Comment:
```suggestion
        allow: [
          ...(viteOverrides?.server?.allow ?? []),
          searchForWorkspaceRoot(process.cwd()),
          binaryRoot,
        ],
        ...(viteOverrides?.server ?? {})
```

Reply:
The reason I'm not doing this everywhere is because of Vite's own `mergeConfig` which manages all of the deeply nested merging.